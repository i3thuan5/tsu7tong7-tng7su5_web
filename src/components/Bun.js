import React from 'react';
import { Block, HanLoTsua, HanLoSu } from 'demo-ui';


const Bun = ({查詢結果}) => (
  <Block>
    {
      查詢結果.map(一句 => (
        <div key={一句.漢字} className='app line'>
          <HanLoTsua 羅馬字={一句.臺羅斷詞} 漢字={一句.漢字} 是否合音={false}/>
          <div>
          {
            一句.白話字.split(' ').map(一詞 =>
              <HanLoSu key={一詞} 漢字={一詞} 羅馬字=''/>
            )
          }
          </div>
        </div>
      ))
    }
  </Block>
);

export default Bun;
