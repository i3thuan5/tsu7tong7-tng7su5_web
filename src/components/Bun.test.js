import React from 'react';
import renderer from 'react-test-renderer';
import { HanLoTsua } from 'demo-ui';
import Bun from './Bun';



it('顯示空結果', () => {
  const tree = renderer
    .create(<Bun 查詢結果={[]}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})


it('顯示一漢羅', () => {
  const tree = renderer
    .create(<Bun 查詢結果={[{漢字: '漢字', 臺羅斷詞: '臺羅斷詞', 白話字: '白話字'}]}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
