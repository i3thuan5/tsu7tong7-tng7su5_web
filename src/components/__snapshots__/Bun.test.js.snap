// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`顯示一漢羅 1`] = `
<div
  className="app block"
>
  <div
    className="app line"
  >
    <div
      className="app text block"
    >
      <ruby
        className="app ruby"
      >
        臺羅斷詞
        <rt>
          漢字
        </rt>
      </ruby>
    </div>
    <div>
      <ruby
        className="app ruby"
      >
        
        <rt>
          白話字
        </rt>
      </ruby>
    </div>
  </div>
</div>
`;

exports[`顯示空結果 1`] = `
<div
  className="app block"
/>
`;
